### PR TITLE
wordle: add filter condition

### PR DESCRIPTION
**Pull request by DanaMC18**
* add `confirmedLetters` array to `WordleGame`
* use `confirmedLetters` to further filter dictionary of possible words

<details>
<summary>Local tests:</summary>

<b>proxy:</b>

```bash
'12972 words left to choose from',
'I guess EYOTS.',
'- - O - -',
'71 words left to choose from',
'I guess GOOKY.',
'- - O - Y',
'13 words left to choose from',
'I guess RHODY.',
'- - O - Y',
'7 words left to choose from',
'I guess FRORY.',
'- R O - Y',
'3 words left to choose from',
'I guess IRONY.',
'- R O - Y',
'1 words left to choose from',
'I guess PROXY.',
'P R O X Y',
'6/6: PROXY'
```

<b>shire:</b>

```bash
'12972 words left to choose from',
'I guess OLENT.',
'- - - - -',
'1669 words left to choose from',
'I guess GASES.',
'- - - - -',
'58 words left to choose from',
'I guess HERSE.',
'- - - - E',
'1 words left to choose from',
'I guess SHIRE.',
'S H I R E',
'4/6: SHIRE'
```

<b>solar:</b>

```bash
'12972 words left to choose from',
'I guess SDEIN.',
'S - - - -',
'498 words left to choose from',
'I guess STAGS.',
'S - - - -',
'67 words left to choose from',
'I guess SAUBA.',
'S - - - -',
'18 words left to choose from',
'I guess SPYAL.',
'S - - A -',
'2 words left to choose from',
'I guess SOLAR.',
'S O L A R',
'5/6: SOLAR'
```
</details>

### CODE REVIEW

File: src/scripts/wordle_guesser.js [hunk 1]
```diff
@@ -21,9 +22,11 @@ class WordleGame {
       if (guess[i] === this.secret[i]) {
         this.answer[i] = char;
         this.possibleLetters[i] = [char];
+        this.confirmedLetters.push(char);
```
Comment (Asterids) [inline src/scripts/wordle_guesser.js:25]: just a thought, might be worth renaming `possibleLetters` and `confirmedLetters` to something like `possibleLettersAtPosition` and `confirmedLettersInWord`? longer but clearer

File: src/scripts/wordle_guesser.js [hunk 2]
```diff
@@ -37,7 +40,8 @@ class WordleGame {
   }
 
   isWordValid(word) {
-    return word.split('').every((char, i) => this.possibleLetters[i].includes(char));
+    return word.split('').every((char, i) => this.possibleLetters[i].includes(char))
+      && this.confirmedLetters.every((char) => word.includes(char));
```
Comment (Asterids) [inline src/scripts/wordle_guesser.js:44]: love your addition! should solve the issue you noticed. it took me a hot second to parse these both tbh, was thinking of pulling these out into named variables but honestly I think the names would be more complicated than the evaluations :joy:

might be a dumb q, but do you have to split the `word` string again before you can use `.includes`?

**Review by Asterids — APPROVED** [review]
left a couple comments but LGTM :+1:

File: src/scripts/wordle_guesser.js (hunk 1 above)
Comment (DanaMC18) [reply src/scripts/wordle_guesser.js]: Renamed!

File: src/scripts/wordle_guesser.js (hunk 2 above)
Comment (DanaMC18) [reply src/scripts/wordle_guesser.js]: Added named variables. They are bad names, but hopefully descriptive?

**Review by Asterids — APPROVED** [review]
Nice!

**Comment by mincer-ray** [comment]
i invited you as a collaborator so you can merge and also not have to fork the repo

**Pull request merged**